examples/CInterface/nonlings_fed2.c

```diff
@@ -56,20 +86,21 @@ int main()
   vfed = helicsCreateValueFederate(fedinfo);
   printf(" Value federate created\n");
 
-  sub = helicsFederateRegisterSubscription(vfed,"testA","double","");
+  sub = helicsFederateRegisterSubscription(vfed,"testA","string","");
   printf(" Subscription registered\n");
 
   /* Register the publication */
-  pub = helicsFederateRegisterGlobalPublication(vfed,"testB","double","");
+  pub = helicsFederateRegisterGlobalPublication(vfed,"testB","string","");
   printf(" Publication registered\n");
 
-
   /* Enter initialization mode */
   status = helicsFederateEnterInitializationMode(vfed);
   printf(" Entered initialization mode\n");
-  
+  double y = 1.0, x = 0, /*xprv = 100,*/ yprv=100;
+  int global_conv=0,my_conv=0,other_conv; /* Global and local convergence */
 
-  status = helicsPublicationPublishDouble(pub, y);
+  snprintf(sendbuf,100,"%18.16f,%d",y,my_conv);
```

Comment:
Oh I see,  to add a convergence flag.  
Why is the convergence flag necessary?  Do you get situations where neither if publishing but it hasn't converged yet?  I would like to understand this a little better to make sure there is no issue with the convergence inside HELICS.

Reply:
Without passing the convergence flag, the federate would not know if the federation has converged. The force_iteration and no_iteration flags are self-explanatory. What does the iterate_if_needed flag do?

Reply:
iterate_if_needed gives helics control over whether to iterate or not.  Basically it will iterate if any values at the current time are received (and the iteration limit has not been exceeded) otherwise it will advance time.

Reply:
maybe it should iterate_if_updated or something better named?